### PR TITLE
Common folder

### PR DESCRIPTION
**Pull request by V-Fries**
1) Creation of a common folder for JWTManager and Revamped JWT manager in order to have a common tool to decode tokens
2) adapt the code in User Management in order to adapt to the new common JWT manager
3) change the docker so that it copies the common folder with 'cp -r'
4) adapt the github action consequently to 3)

### CODE REVIEW

**Review by tdameros — APPROVED** [review]

**Review by aLeuleu — APPROVED** [review]
LGTM

File: common/src/jwt_managers.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                 private_key: str | None,
+                 public_key: str | None,
+                 algorithm: str,
+                 expiration_time_minutes: int | None):
+        self.private_key = private_key
+        self.public_key = public_key
+        self.algorithm = algorithm
+        self.expiration_time_minutes = expiration_time_minutes
+
+    def generate_token(self, payload_arg: dict) -> (bool, str | None, list[str] | None):
```
Comment (EthanDelage) [inline common/src/jwt_managers.py:19]: In my opinion, this function should be renamed `generate_jwt()` to keep the same syntax.

File: user_management/src/user_management/settings.py [hunk 2]
```diff
@@ -157,7 +148,7 @@
 # https://docs.djangoproject.com/en/4.2/ref/settings/#databases
 
 # #SQLite 3 for debugs and tests
-DEBUG = False
+DEBUG = True
```
Comment (EthanDelage) [inline user_management/src/user_management/settings.py:151]: Do not let `DEBUG=True` in production

**Review by EthanDelage — CHANGES_REQUESTED** [review]

File: user_management/src/user_management/settings.py (hunk 2 above)
Comment (V-Fries) [reply user_management/src/user_management/settings.py]: https://github.com/tdameros/42-transcendence/pull/96/commits/deb603477bb8f058ae2f03f3690e7c6ab06f18a5

File: common/src/jwt_managers.py (hunk 1 above)
Comment (V-Fries) [reply common/src/jwt_managers.py]: https://github.com/tdameros/42-transcendence/pull/96/commits/deb603477bb8f058ae2f03f3690e7c6ab06f18a5

**Review by EthanDelage — APPROVED** [review]

**Review by hferraud — APPROVED** [review]

**Pull request merged**